app/src/main/java/org/openobservatory/ooniprobe/fragment/measurement/FailedFragment.java

```diff
@@ -56,12 +54,12 @@ public static FailedFragment newInstance(Measurement measurement) {
 		AbstractSuite testSuite = failedMeasurement.result.getTestSuite();
 		testSuite.setTestList(abstractTest);
 		testSuite.setResult(failedMeasurement.result);
-		failedMeasurement.setReRun(getContext());
 
 		RunningActivity.runAsForegroundService((AbstractActivity) getActivity(),
 				testSuite.asArray(),
 				() -> {
 					try {
+						failedMeasurement.setReRun(getContext());
 						getActivity().finish();
```

Comment:
What does `getActivity().finish()` do? Do we perhaps want to mark it as re-run after also that operation is successful?

Reply:
`getActivity().finish(); ` closes the details page for the item and returns to the list.

`failedMeasurement.setReRun(getContext()); `  marks the measurement as `re-run` among other things.